crates/oxc_linter/src/loader/partial_loader/astro.rs

```diff
@@ -42,8 +42,9 @@ impl<'a> AstroPartialLoader<'a> {
             return None;
         };
 
-        let js_code =
-            Span::new(start + ASTRO_SPLIT.len() as u32, end).source_text(self.source_text);
+        // move start to the end of the ASTRO_SPLIT
+        let start = start + ASTRO_SPLIT.len() as u32;
```

Comment:
This logic looks identical. Am I missing something?

Reply:
yep, the issue here is that the `start` used for getting the source text had the length added to it, but then the `start` without the addition was stored as the start offset